code-for-canada/project-thundercat
# [Edit Email] Edit email no save

# Description

Added functionality for opening email response in a modify dialog with all inputs defaulted to the saved values.

NOTE: This PR does not save the existing email, but instead saves a new email. This will is fixed in PR #192, which will be merged into this one:
Editing an single email results in this:
![image](https://user-images.githubusercontent.com/2746350/55814348-4f29c600-5abc-11e9-8d0e-1bb8821b95e6.png)
![image](https://user-images.githubusercontent.com/2746350/55814362-594bc480-5abc-11e9-8888-16a9ee273b61.png)



## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- Breaking change (it "adds" a new email rather than editing an existing one; will is fixed in PR #192 merged into this one before the merge to master)

## Screenshot

Open Add Email Response
![image](https://user-images.githubusercontent.com/2746350/55813687-310f9600-5abb-11e9-8151-7c0adbb06fef.png)

Fill out the form and save it:
![image](https://user-images.githubusercontent.com/2746350/55813752-569c9f80-5abb-11e9-931f-37c4674e56ca.png)
![image](https://user-images.githubusercontent.com/2746350/55813789-69af6f80-5abb-11e9-9959-f71227951f48.png)

Edit the email:
![image](https://user-images.githubusercontent.com/2746350/55813884-8c418880-5abb-11e9-9f3c-d3684583b181.png)
![image](https://user-images.githubusercontent.com/2746350/55813911-9794b400-5abb-11e9-9901-428e56f3006a.png)

Response is populated with the previous values:
![image](https://user-images.githubusercontent.com/2746350/55814090-e3dff400-5abb-11e9-96e9-df3030734295.png)
![image](https://user-images.githubusercontent.com/2746350/55814113-ee01f280-5abb-11e9-8f5c-5b5bf8ed8797.png)

Edit the response:
![image](https://user-images.githubusercontent.com/2746350/55814152-0114c280-5abc-11e9-8f76-5dbfc1503aa5.png)
![image](https://user-images.githubusercontent.com/2746350/55814173-0b36c100-5abc-11e9-9b69-f88d5d904fed.png)

Expand new email and check that contents were correctly changed:
![image](https://user-images.githubusercontent.com/2746350/55814254-286b8f80-5abc-11e9-8b65-8fa45fb7f388.png)
![image](https://user-images.githubusercontent.com/2746350/55814301-37ead880-5abc-11e9-873e-7795c75ea117.png)



# Testing

Applicable for all code changes.

Manual steps to reproduce this functionality:

1.  Start the eMIB, navigate through the pre-test instructions, open the inbox tab
2.  Open any email
3.  Click "Add Email Response"
4.  Fill out the form
5.  Click "Save Response"
6.  Expand the "Email Response"
7.  Click "Edit Response"
8.  Make any desired changes
9. Click "Save Response"
10. Verify that the second "Email Response" has the changes

# Checklist

Applicable for all code changes.

~~- [ ] I have made corresponding changes to the documentation~~
